src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -283,21 +304,19 @@ private UnboundLambda BindAnonymousFunction(AnonymousFunctionExpressionSyntax sy
             var data = lambda.Data;
             if (data.HasExplicitlyTypedParameterList)
             {
+                int firstDefault = -1;
                 for (int i = 0; i < lambda.ParameterCount; i++)
                 {
-                    // UNDONE: Where do we report improper use of pointer types?
-                    var type = data.ParameterTypeWithAnnotations(i).Type;
-                    if (type is { })
+                    // paramSyntax should not be null here; we should always be operating on an anonymous function which will have parameter information
```

Comment:
>paramSyntax should not be null here [](http://example.com/codeflow?start=23&length=35)

Let's add `Debug.Assert(paramSyntax is { });`, to avoid needing `paramSyntax!` below.